src/components/views/rooms/RoomSettings.js

```diff
@@ -55,9 +55,20 @@ module.exports = React.createClass({
             power_levels_changed: false,
             tags_changed: false,
             tags: tags,
-            areNotifsMuted: areNotifsMuted
+            areNotifsMuted: areNotifsMuted,
         };
     },
+
+    componentWillMount: function() {
+        var self = this;
+        MatrixClientPeg.get().getRoomVisibility(
+            this.props.room.roomId
+        ).done((result) => {
+            self.setState({ isRoomPublished: result.visibility === "public" });
```

Comment:
Nit: You're being awesome and using arrow functions, but yet still using `self`. Arrow functions capture `this` correctly, so you can drop the `self`.